File: parent-337/pinyougou-web-manager/src/main/webapp/js/service/itemCatService.js
app.service('itemCatService', function ($http) {
    this.findByParentId = function (parentId) {
        return $http.get('../itemCat/findByParentId.do?parentId=' + parentId);
    };

    this.add = function (entity) {
        return $http.post('../itemCat/add.do', entity);
    };

    this.findOne = function (id) {
        return $http.get('../itemCat/findOne.do?id=' + id);
    };

    this.update = function (entity) {
        return $http.post('../itemCat/update.do', entity);
    };

    this.del = function (selectIds) {
        return $http.get('../itemCat/del.do?selectIds=' + selectIds);
    };
});

File: parent-337/pinyougou-interface/src/main/java/cn/ithcast/core/service/ItemCatService.java
package cn.ithcast.core.service;

import cn.ithcast.core.pojo.item.ItemCat;
import entity.Result;

import java.util.List;

public interface ItemCatService {

    /**
     * 根据父id进行查询
     *
     * @param parentId
     * @return
     */
    List<ItemCat> findByParentId(Long parentId);

    /**
     * 添加商品分类
     *
     * @param itemCat
     */
    void add(ItemCat itemCat);

    /**
     * 修改商品分类之数据回显
     *
     * @param id
     * @return
     */
    ItemCat findOne(Long id);

    /**
     * 修改商品
     *
     * @param itemCat
     */
    void update(ItemCat itemCat);

    /**
     * 删除商品分类
     *
     * @param selectIds
     * @return
     */
    Result del(Long[] selectIds);

    /**
     * 查询所有
     * @return
     */
    List<ItemCat> findAll();
}
